### PR TITLE
[4.3] Guided Tours remove broken code for creation of basic step

### PR DESCRIPTION
**Pull request by heelc29**
### Summary of Changes

Fix the creation of the basic step when creating a new tour
https://github.com/joomla/joomla-cms/blob/338a9402a2702fa5d30bd16acf61138558bcf18f/administrator/components/com_guidedtours/src/Model/TourModel.php#L82-L95

### Testing Instructions

Create a new tour

### Actual result BEFORE applying this Pull Request

The new tour does not contain steps

database errors:
image
image
image
image
image

### Expected result AFTER applying this Pull Request

The tour saves properly, and when going to the steps page, the user is prompted to create a step.

### Link to documentations
Please select:

- [x] No documentation changes for docs.joomla.org needed
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by richard67** [comment]
@heelc29 I don't get these database errors when creating a new tour on a clean, current 4.3-dev branch after a new installation of Joomla.

Could you describe in detail what you have done to get these errors? And what's your PHP and your database server type (MySQL or MariaDB) and version?

**Comment by heelc29** [comment]
> Could you describe in detail what you have done to get these errors?

@richard67 I get it while debugging (xDebug) why no step is created. The exception is caught by the table class but you can check it also with this code
```php
if (!$table->store()) {
    throw new \Exception(implode("\n", $table->getErrors()), 500);
}
```
image

> And what's your PHP and your database server type (MySQL or MariaDB) and version?

image

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on ecc74a58c05680669b10b432148a03ddb798e0c0<br /><br />I can confirm that without this PR, the basic step doesn't get created when using "save" or "save and new", while it gets created with this PR applied.

I also can confirm that the columns handled by this PR here indeed don't have default values, so they should be set when creating a new record.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40297">issues.joomla.org/tracker/joomla-cms/40297</a>.</sub>

File: administrator/components/com_guidedtours/src/Model/TourModel.php [hunk 1]
```diff
@@ -90,6 +90,11 @@ public function save($data)
             $table->description = '';
             $table->tour_id     = $tourId;
             $table->published   = 1;
+            $table->position    = 'center';
+            $table->target      = '';
+            $table->type        = 0;
+            $table->url         = '';
+            $table->language    = '*';
```
Comment (heelc29) [inline administrator/components/com_guidedtours/src/Model/TourModel.php]: @richard67 Should I use `$lang`/`$data['language']` here?

Comment (richard67) [reply administrator/components/com_guidedtours/src/Model/TourModel.php]: I don't know. @khu5h1 @obuisard What do you think?

Comment (obuisard) [reply administrator/components/com_guidedtours/src/Model/TourModel.php]: The language has to be the language of the tour

**Comment by obuisard** [comment]
Now, I am not sure we really need a basic step here....

**Comment by heelc29** [comment]
> Column 'ordering' cannot be null

This column should have a default value 🤔

https://github.com/joomla/joomla-cms/blob/3209f684698432ee8e498fde269876de6c0ffcb4/installation/sql/mysql/extensions.sql#L986


> Now, I am not sure we really need a basic step here....

@obuisard We can also delete the broken code ...

**Comment by obuisard** [comment]
I would just remove the code that is never executed rather than try fixing it. To me, it does not help to have a pre-installed step when a tour is created.
Right now, the behavior users get is that no step is created when creating a new tour. 
We should just leave it like that (the behavior) and remove the code that was supposed to create a new step but is never reached.

**Comment by heelc29** [comment]
> remove the code that was supposed to create a new step but is never reached

@obuisard PR updated

**Comment by obuisard** [comment]
> > remove the code that was supposed to create a new step but is never reached
> 
> @obuisard PR updated

Thank you so much @heelc29, really appreciated.

**Comment by khu5h1** [comment]
I have tested this item :white_check_mark: successfully on 0ce5c6b9fc8bdd164748336c0b0b93d4aa0fc41e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40297">issues.joomla.org/tracker/joomla-cms/40297</a>.</sub>

**Comment by obuisard** [comment]
I have tested this item :white_check_mark: successfully on 0ce5c6b9fc8bdd164748336c0b0b93d4aa0fc41e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40297">issues.joomla.org/tracker/joomla-cms/40297</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40297">issues.joomla.org/tracker/joomla-cms/40297</a>.</sub>

**Comment by sdwjoomla** [comment]
Thanks @heelc29

**Pull request merged**